src/transforms/kotlin/svgToDrawable.ts

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import path, { dirname } from "path";
+import fs from "fs-extra";
+import { Transform } from "style-dictionary/types/Transform";
+import svg2vectordrawable from "svg2vectordrawable";
+import _ from "lodash";
+
+/**
+ * A transformer to change svg path to vector drawable path
+ * Also generates the drawable.
+ */
+export default {
+  type: "value",
+  matcher: function (token) {
+    return token.type === "icon";
+  },
+  transformer: function (token, platform) {
+    const iconPath = path.join(dirname(require.main!.filename), token.value);
+    const resPath = `/res/drawable`;
+
+    // Snake case and replace `icon` with `ic` as this is the convention on Android
+    // and on Material
+    const imageId = _.snakeCase(token.name.replace("icon", "ic"));
```

Comment:
Just a small non-blocking remark, maybe replace `icon` by `ic` only if `token.name` starts by `icon` (and replace only once).

Reply:
Yup, this code is not really self explanatory I have to say... but all icons will be namespaced with `icon` at the start, and `String.prototype.replace` only changes the first occurence. 

But i have to admit this is not the most solid piece of code i've ever written